### DISCUSSION TITLE
4 advice to resty

### DISCUSSION THREAD

**Discussion by Nyx2022** [go-resty/resty#513] [Ideas]
i used resty for long time,but found 2 features or bugs need to be implemented
1. i hope resty will supprt concurrent safe
2. i hope resty can handle special encoding,like this photo
![不支持特俗编码](https://user-images.githubusercontent.com/51039208/149131272-60f94022-6fa7-4a3e-b699-775974854b47.png)
sometimes i actually need some special encoding，but resty can't deal with it 
3. resty.Request can set proxy，now only resty.Client can set proxy,i hope resty.request can support it too
4. more easy way to set redirect policy,i think the way now to set redirect policy is a little difficultto call

**Comment by jeevatkm** [comment]
@Nyx2022 Please take a look at my response.
1. I believe you're talking about the client instance; if yes, I plan to add read-write mutex to the client instance in the v3.0.0 goal.
2. I plan to add a mechanism for Resty user to add their parsing logic in the v3.0.0 goal.
3. Request level proxy is not feasible to achieve; it will have an impact on the client 
4. Could you provide details of the difficulty in using the redirect policy?
